### ISSUE TITLE
CaffeModel_Trained_on_ImageNet

### ISSUE THREAD

**Issue by liu666666** [liuzhuang13/DenseNetCaffe#7]
Hi,Thanks for your sharing !
I noticed that your team have released  the torch model trained on the ImageNet, Could you please release your caffe model trained on the ImageNet?

**Comment by liuzhuang13** [comment]
Thanks for your interest! Sorry we have no pre-trained models on ImageNet in caffe currently. 

If you really want to use the models, maybe you can try to convert the torch model into caffe models by using https://github.com/facebook/fb-caffe-exts#torch2caffe . I tried but I didn't succeed.

We'll make the caffe pretrained models as soon as possible. Thanks

**Comment by sunformoon** [comment]
@liu666666 Hi, have you succeeded on transferring the trained caffe model from torch version?
Thank you!

**Comment by liu666666** [comment]
No, sorry.

**Comment by kaishijeng** [comment]
Liu,

   Are imagenet of solver,prototxt, train_densenet.prototxt and test_densenet.prototxt available?

Thanks,

**Comment by szq0214** [comment]
@kaishijeng Here are the solver.prototxt and train_val_densenet.prototxt files for densenet_121. I think you can use them to train caffe models on ImageNet without any modification. @liu666666 @sunformoon We will release the pre-trained models soon.
[densenet_121.zip](https://github.com/liuzhuang13/DenseNetCaffe/files/706566/densenet_121.zip)

**Comment by kaishijeng** [comment]
Thanks and will try it

**Comment by liuzhuang13** [comment]
@sunformoon @liu666666 @kaishijeng Sorry for the late response. Our pretrained DenseNet-121 and prototxt in Caffe has just been released here https://github.com/liuzhuang13/DenseNet#imagenet-and-pretrained-models

Thanks for your interest!

**Comment by saiguruju** [comment]
Hi, could you share the solver and train prototxt files of Densenet(L=40, K=12) for training cifar100, I have found these files for cifar10, training the same for cifar100 have given me low accuracies. 
Thanks!!

**Comment by liuzhuang13** [comment]
@saiguruju  Thanks for your interest. Sorry we didn't test this caffe code for cifar100, but in our original experiment on paper, we use exactly the same setting as cifar10 when training on cifar 100. 

I've heard other people also report the lower accuracy of cifar100 than reported on our paper, but sorry I don't know the reasons either. Possibly the difference in accuracy from our original paper is due to some internal differences between Torch and caffe. This is also the case for ImageNet dataset, see our discussion here https://github.com/liuzhuang13/DenseNet/issues/10

**Comment by saiguruju** [comment]
By same setting, you mean that prototxt files (solver and train) will be same, and accuracy might be little less. I have only trained for 20000 iterations for same setting with hingeloss(L2) on cifar100, accuracy went up to 0.05, should I train upto 230000 and observe? 
I just wanted to know if its getting trained in the right way.

Thanks for prompt reply!

**Comment by liuzhuang13** [comment]
@saiguruju  The solver should be the same, but at least you should modify the train.prototxt to 100 classes instead of 10.

If you only get an accuracy of 0.05 after 20k iterations, there ought to be bugs in your code... I think at least you should get 60% accuracy if there are no bugs, even where there is no data augmentation.

**Comment by saiguruju** [comment]
I did have 100 classes. It is possible that my code has gone wrong somewhere!
Thank you!

**Comment by liuzhuang13** [comment]
@saiguruju We used softmax loss throughout the work, maybe the problem is your L2 loss. This is just my guess

**Comment by saiguruju** [comment]
@liuzhuang13  Thanks! It worked the way you predicted with softmax, there is some problem with L2 loss.
